### DISCUSSION TITLE
ModuleNotFoundError: No module named 'torch._C'

### DISCUSSION THREAD

**Discussion by chconnor** [openai/whisper#84] [Q&A] [answered]
Hi -- installation went fine, no errors. I'm getting a "ModuleNotFoundError: No module named 'torch._C'" error when I try to run it. Same thing if I try to "import whisper" on the python console. "import torch" works fine when I try that.

```
$ whisper
Traceback (most recent call last):
  File "/home/casey/.local/bin/whisper", line 5, in <module>
    from whisper.transcribe import cli
  File "/home/casey/.local/lib/python3.10/site-packages/whisper/__init__.py", line 11, in <module>
    from .audio import load_audio, log_mel_spectrogram, pad_or_trim
  File "/home/casey/.local/lib/python3.10/site-packages/whisper/audio.py", line 8, in <module>
    import torch.nn.functional as F
  File "/home/casey/.local/lib/python3.10/site-packages/torch/nn/__init__.py", line 1, in <module>
    from .modules import *  # noqa: F403
  File "/home/casey/.local/lib/python3.10/site-packages/torch/nn/modules/__init__.py", line 1, in <module>
    from .module import Module
  File "/home/casey/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 7, in <module>
    from ..parameter import Parameter
  File "/home/casey/.local/lib/python3.10/site-packages/torch/nn/parameter.py", line 2, in <module>
    from torch._C import _disabled_torch_function_impl
ModuleNotFoundError: No module named 'torch._C'
```

Any thoughts? Thanks!

Edit: some more detail:
Python 3.10.6

While "import torch" works, "from torch._C import *" gives the same error as above.

~/.local/lib/python3.10/site-packages/torch has no _C.* file which [this](https://github.com/pytorch/pytorch/issues/574) leads me to believe it should...

**Comment by aakashsengar** [comment]
Use python version 3.9.9

**Reply by DimitarVanguelov** [reply]
Sorry but "use python version 3.9.9" is not a good reply. Per the README:
> but the codebase is expected to be compatible with Python 3.7 **or later**

Nevertheless, I tried using Python 3.9.9 and still got the same error as OP. I originally also tried with 3.10.6, saw this issue, tried 3.9.9 and no improvement.

**Reply by aakashsengar** [reply]
For me upgrading to python3.9.9 worked.

**Comment by tigvarts** [comment, marked as answer]
Ran into the same thing recently. Removing torch and then reinstalling it through pip as recommended on pytorch.org solved the issue for me.
P.S. Used ubuntu 22.04, python 3.8, virtualenv, pip. I suspect that not having GPU on my laptop could be the cause for this issue, but I'm completely not sure about it.

**Reply by ndvbd** [reply]
@tigvarts same problem, same envirnoment, same solution. Thanks. It's strange that uninstalling torch and reinstalling solves the issue...

**Reply by kanishkbh** [reply]
same problem with python 3.10.6, Ubuntu 22.04, virtualenv, pip, no GPU. reinstalling worked.

**Comment by JoseLuiz432** [comment]
Encountered the same issue today with Ultralytics. What worked for me was:

- Uninstall torch, torchvision, and Ultralytics.
```bash
pip uninstall torch torchvision ultralytics
```
- Visit the main page of [Pytorch](https://pytorch.org/) and scroll down until  INSTALL PYTORCH section.
- install using: 
```bash
 pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu
```
